src/service.ts

```diff
@@ -331,6 +332,39 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
+      if (file instanceof Directory) {
+        let folder: any;
+        if (!(file instanceof Project)) {
+          folder = zipFile.folder(file.name);
+        }
+        file.mapEachFile((file: File) => defineZip(file, folder));
+      } else {
+        if (!isUndefined(folder)) {
+          folder.file(file.name, file.data);
+        } else {
+          zipFile.file(file.name, file.data);
+        }
+      }
+    }
+    defineZip(project);
+    zipFile.generateAsync({type: "blob", mimeType: "application/zip"}).then((blob: any) => {
+      const link = document.createElement("a");
+      link.download = zipName;
+      link.href = URL.createObjectURL(blob);
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(document.body.lastChild);
```

Comment:
can you change it to `document.body.removeChild(link);` and add a comment why we are doing it, including why we are creating a-tag in first place?

Reply:
Yeah, I was doing this...

Reply:
actually, can you extract entire code and make a function from it?